### PR TITLE
Qcforward add perf

### PR DESCRIPTION
**Pull request by jcrivenaes**
Many changes, structural changes

### CODE REVIEW

File: docs/qcforward.rst [hunk 1]
```diff
@@ -35,12 +38,18 @@ Common fields (same input inside or outside RMS)
...
+perflog:
+  The name of the perforation log. A dictionary with keys ``name``, ``range``,
+  see examples (optional). If present, zonelog matching will be performed only
+  in perforation intervals.
 
-zonelogrange
-  A list with two entries, defining minimum and maximum zone to use (both ends
-  are inclusive). Default is [1, 99]. It is recommended to set range explicitly.
+well_resample:
+  To speed up calulations (but on cost of less precision), the wells are resmapled
```
Comment (HansKallekleiv) [inline docs/qcforward.rst:50]: typo

File: somefile.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+actions_all: {stopthreshold: 20, warnthreshold: 80}
```
Comment (HansKallekleiv) [inline somefile.yml:1]: Should this file be here?

File: src/fmu/tools/qcforward/_qcforward.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
+"""The _qcforward module contains the base class"""
+from __future__ import absolute_import, division, print_function  # PY2
```
Comment (HansKallekleiv) [inline src/fmu/tools/qcforward/_qcforward.py:2]: Remove?

**Review by HansKallekleiv — APPROVED** [review]

File: src/fmu/tools/qcforward/_qcforward_data.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+            rmswells = [wll.name for wll in self._project.wells]
+            CMN.print_debug("All RMS wells: {}".format(rmswells))
+
+            CMN.print_debug("Data wells to match: {}".format(wnames))
+
+            for rmswell in rmswells:
+                for wreg in wnames:
+                    CMN.print_debug("Trying match {} vs re {}".format(rmswell, wreg))
+                    if re.match(wreg + "$", rmswell):
```
Comment (tnatt) [inline src/fmu/tools/qcforward/_qcforward_data.py:277]: It looks like a well will be loaded multiple times if it matches multiple regexes, maybe we should write something like:

```
if any(re.match(wreg + "$", rmswell) for wreg in wnames):
```

**Review by tnatt — APPROVED** [review]
In general it looks good, only small comments from me.

I would also like to see BlockedWells added to the data parsing.

File: src/fmu/tools/qcforward/_qcforward_data.py (hunk 4 above)
Comment (jcrivenaes) [reply src/fmu/tools/qcforward/_qcforward_data.py:277]: Good point

File: src/fmu/tools/qcforward/_qcforward_data.py [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+                )
+                gprops.append(current)
+
+        self._gridprops = xtgeo.GridProperties()
+        self._gridprops.append_props(gprops)
+
+    def read_wells(self):  # pylint: disable=too-many-branches
+        """Reading wells"""
+
+        if "wells" in self._reuse:
```
[the remark this answers is not in the document]
Comment (jcrivenaes) [reply src/fmu/tools/qcforward/_qcforward_data.py:230]: Should be possible but requires some extra coding/tests. Could you add this as an issue?

File: src/fmu/tools/qcforward/_qcforward_data.py [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+        """dict:: Report mode settings"""
+        return self._reportmode
+
+    @property
+    def nametag(self):
+        """str:: Nametag for data set"""
+        return self._nametag
+
+    @staticmethod
+    def _unpack_dict1(mydict):
```
[the remark this answers is not in the document]
Comment (jcrivenaes) [reply src/fmu/tools/qcforward/_qcforward_data.py:94]: I used it in a check, but then refactored. Leave it there for a while...

File: src/fmu/tools/qcforward/_qcforward_data.py [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+        self.read_gridprops()
+        self.read_wells()
+
+    def set_path(self):
+        """General path prefix settings"""
+        self._path = self._data.get("path", ".")
+
+    def set_report(self):
+        """General report settings"""
+        myreport = self._data.get("report", {"file": "default.yml", "mode": "write"})
```
[the remark this answers is not in the document]
Comment (jcrivenaes) [reply src/fmu/tools/qcforward/_qcforward_data.py:151]: Yes should be csv ... a typo

File: somefile.yml (hunk 2 above)
Comment (jcrivenaes) [reply somefile.yml:1]: nope, will remove

File: src/fmu/tools/qcforward/_qcforward_data.py [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+
+        self._reportfile = myreport["file"]
+        self._reportmode = myreport["mode"]
+
+    def set_nametag(self):
+        """General nametag settings"""
+        self._nametag = self._data.get("nametag", {"file": "unset_nametag"})
```
[the remark this answers is not in the document]
Comment (jcrivenaes) [reply src/fmu/tools/qcforward/_qcforward_data.py:158]: Well spotted

File: src/fmu/tools/qcforward/_qcforward_data.py (hunk 4 above)
Comment (jcrivenaes) [reply src/fmu/tools/qcforward/_qcforward_data.py:278]: Will test in next PR

**Pull request merged**